frb_rust/src/ffi_binding/mod.rs

```diff
@@ -44,3 +44,23 @@ fn vec_resize(vec: &mut Vec<u8>, new_len: i32) {
 pub unsafe extern "C" fn frb_rust_vec_u8_free(ptr: *mut u8, len: i32) {
     vec_from_leak_ptr(ptr, len);
 }
+
+#[cfg(not(target_family = "wasm"))]
+pub mod shutdown {
+    use std::ffi::c_void;
+
+    use allo_isolate::{store_dart_post_cobject, ffi::DartCObject};
+
+    /// Called by Dart's NativeFinalizer on isolate group shutdown.
+    #[unsafe(no_mangle)]
+    pub unsafe extern "C" fn frb_shutdown_callback(_: *mut c_void) {
+
+        unsafe extern "C" fn devnull(_: i64, _: *mut DartCObject) -> bool {
+            true
```

Comment:
hmm, shall we do `true` or `false` here

From https://github.com/dart-lang/sdk/blob/6ad56c68dc01466d1a51fbed1ac60fd82ca648f6/runtime/include/dart_native_api.h#L115:

> If false is returned, the message was not enqueued and ownership of external typed data in the message remains with the caller

It seems false is a bit more reasonable, since we really mean the message remains with the caller

Reply:
Not sure about this. While returning `false` is technically correct that would make `StreamSink::add` to error and that might trigger some logging, requests to error tracking services, maybe there will even be some panic in user code. And at this point application is shutting down and we just want it to die silently.

I've added doc comment to explain this. Can change it to return `false` if you want, that's not a big deal.

Reply:
Ah I see, good point! But if we have true, then we are leaking memory...

Reply:
> But if we have true, then we are leaking memory...

Well, that's also does not really matter since memory will be reclaimed anyway when shutdown is completed. 

I think it't ready for review, but let me know if you want this changed.